packages/aws-cdk-lib/aws-elasticloadbalancingv2/lib/alb/application-load-balancer.ts

```diff
@@ -186,6 +186,11 @@ export class ApplicationLoadBalancer extends BaseLoadBalancer implements IApplic
     });
 
     this.ipAddressType = props.ipAddressType ?? IpAddressType.IPV4;
+
+    if (props.ipAddressType === IpAddressType.DUAL_STACK_WITHOUT_PUBLIC_IPV4 && !props.internetFacing) {
+      throw new Error('dual-stack without public IPv4 address can only be used with internet-facing scheme.');
```

Comment:
do you have the link of the document about this restriction?

Reply:
@pahud 
Thank you for your comment. 
That restriction is not mentioned in the [documentation](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#ip-address-type), but it is stated on the management console.

<img width="727" alt="image" src="https://github.com/aws/aws-cdk/assets/13781813/6f722cc8-cf82-46ff-89bf-53fbc32b0e45">

If the internal scheme is selected, the "Dual-stack without public IPv4" option disappears and cannot be configured.

<img width="904" alt="image" src="https://github.com/aws/aws-cdk/assets/13781813/825b9169-a714-4a09-8422-9152b03c729c">

Reply:
Awesome!